src/main/java/mate/academy/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,155 @@
+package mate.academy.internet.shop.dao.jdbc;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internet.shop.dao.ShoppingCartDao;
+import mate.academy.internet.shop.exceptions.DataProcessingException;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Product;
+import mate.academy.internet.shop.model.ShoppingCart;
+import mate.academy.internet.shop.util.ConnectionUtil;
+import org.apache.log4j.Logger;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    private static final Logger LOGGER = Logger.getLogger(ShoppingCartDaoJdbcImpl.class);
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        String query = "INSERT INTO internet_shop.shopping_carts(user_id) values(?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, shoppingCart.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong(1));
+            }
+            LOGGER.info("Shopping cart with id=" + shoppingCart.getId() + " was created");
+            return shoppingCart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creation of shopping cart is failed", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * from internet_shop.shopping_carts_products where cart_id=?;";
```

Comment:
you are selecting a lot of redundant rows here. In shopping_carts_products could be a lot of rows with this shoppingCartId, but you will use only one. Can we select here from shopping_carts? This way we will get only one row.

Reply:
done